code/modules/mob/living/simple_animal/hostile/venus_human_trap.dm

```diff
@@ -245,4 +255,41 @@
 	for(var/i in vines)
 		qdel(i)
 	return ..()
+
+/mob/living/simple_animal/hostile/venus_human_trap/UnarmedAttack(atom/attack_target, proximity_flag, list/modifiers)
+	if(!istype(attack_target, /obj/structure/spacevine))
+		return ..()
+	if(!proximity_flag)
+		return
+	var/obj/structure/spacevine/attacked_spacevine = attack_target
+	if(help_grow <= world.time)
+		help_grow = world.time + 1 SECONDS
+		if(attacked_spacevine.energy >= 2)
+			attacked_spacevine.spread()
+			to_chat(src, span_notice("You help [attacked_spacevine] expand..."))
+		else
+			attacked_spacevine.grow()
+			to_chat(src, span_notice("You help [attacked_spacevine] grow..."))
+	var/turf/vine_turf = get_turf(attack_target)
+	var/list/break_list = list(
+		/obj/machinery/door,
+		/obj/structure/table,
+		/obj/structure/window,
+		/obj/structure/girder,
+		/obj/structure/grille,
+		/obj/structure/rack,
+		/obj/structure/door_assembly,
+		/obj/structure/closet,
+	)
+	for(var/check_contents in vine_turf.contents)
+		if(isliving(check_contents) && !istype(check_contents, /mob/living/simple_animal/hostile/venus_human_trap))
+			UnarmedAttack(check_contents)
+		if(is_type_in_list(check_contents, break_list))
+			UnarmedAttack(check_contents)
```

Comment:
Should be more efficient using a static typecache, it seems.
Please test this to make sure it works, I haven't actually tested to make sure the syntax was entirely correct.
```suggestion
	var/static/list/break_list = typecacheof(list(
		/obj/machinery/door,
		/obj/structure/table,
		/obj/structure/window,
		/obj/structure/girder,
		/obj/structure/grille,
		/obj/structure/rack,
		/obj/structure/door_assembly,
		/obj/structure/closet
	))
	for(var/check_contents in vine_turf.contents)
		if(isliving(check_contents) && !istype(check_contents, /mob/living/simple_animal/hostile/venus_human_trap))
			UnarmedAttack(check_contents)
		if(is_type_in_typecache(check_contents, break_list))
			UnarmedAttack(check_contents)
```